src/jdk.compiler/share/classes/com/sun/tools/javac/code/Types.java

```diff
@@ -5024,6 +5024,55 @@ public Type visitCapturedType(CapturedType t, S s) {
     }
     // </editor-fold>
 
+    // <editor-fold defaultstate="collapsed" desc="Unconditionality">
+    /** Check unconditionality between any combination of reference or primitive types.
+     *
+     *  Rules:
+     *    an identity conversion
+     *    a widening reference conversion
+     *    a widening primitive conversion (delegates to `checkUnconditionallyExactPrimitives`)
+     *    a boxing conversion
+     *    a boxing conversion followed by a widening reference conversion
+     *
+     *  @param source     Source primitive or reference type
+     *  @param target     Target primitive or reference type
+     */
+    public boolean checkUnconditionallyExact(Type source, Type target) {
+        if (isSameType(source, target)) {
+            return true;
+        }
+
+        if (target.isPrimitive()) {
+            return (source.isReference() && isSubtype(source, target)) ||
```

Comment:
I'm a bit confused here - in here we have:
* `target.isPrimitive()`
* `source.isReference()`

So, isn't it the case that `isSubtype(source, target)` is always false?

Reply:
Yes! Thanks for spotting. https://github.com/openjdk/jdk/pull/15638/commits/0fcdca8229c0db81fe3503770ed155a6aaa4f250